trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -886,6 +892,65 @@ private ExpressionResult getNondetStringOrNull(final ILocation loc) {
 		return builder.build();
 	}
 
+	private Result handleVaStart(final IDispatcher main, final IASTFunctionCallExpression node, final ILocation loc,
+			final String name) {
+		final IASTInitializerClause[] arguments = node.getArguments();
+		checkArguments(loc, 2, name, arguments);
+		final ExpressionResultBuilder builder = new ExpressionResultBuilder();
+		final ExpressionResult arg0 =
+				mExprResultTransformer.transformDispatchDecaySwitchRexBoolToInt(main, loc, arguments[0]);
+		builder.addAllExceptLrValue(arg0);
+		// TODO: We do not actually use this parameter, this could be unsound.

```

Comment:
Is there a particular reason?

Reply:
Yes. `va_start(v, p)` has a strange semantic. All parameters after `p` are considered vargars and are therfore copied to `v`. Usually you just use the last parameter before `...` for `p` and I am neither sure, what happens otherwise, nor how we could handle it, since we would need to refer to the function signature that we don't have here.

Reply:
So I just checked it: The compiler does not allow `va_start` for fixed-argument-functions. But if you have a function `f(int a, int b, ...)` and you call `__builtin_va_start(v, a)`, you still get only the `...`-arguments. The C-standard is also not quite precise about that:
>The parameter parmN is the identifier of the rightmost parameter in the variable parameter list in the function definition (the one just before the , ...).
What is supposed to happen, if parmN is not the rightmost parameter? So I think, it is okay, how we currently handle it.

Reply:
:+1: Sounds good. It seems like all other cases are unspecified, so hopefully they shouldn't occur.

Reply:
Okay, so I will replace the TODO with a comment instead.